src/uadatetime.cpp

```diff
@@ -118,3 +118,8 @@ UaString UaDateTime::toString() const
 
         return UaString(result.str().c_str());
 }
+
+OpcUa_Int32 UaDateTime::secsTo(const UaDateTime& dt) const
+{
+	return (dt.m_dateTime - m_dateTime) / pow(10, 7);
```

Comment:
Observation: it's a bit suspicious mix of integer and floating-point arithmetics, these sometimes work as expected and sometimes not (a lot was written on it ...). plus putting a constant into runtime expression - perhaps not necessary?

(e.g. see https://www.geeksforgeeks.org/power-function-cc/ 
`Working of pow() function with integers

The pow() function takes ‘double’ as the arguments and returns a ‘double’ value. This function does not always work for integers. One such example is pow(5, 2). When assigned to an integer, it outputs 24 on some compilers and works fine for some other compilers. But pow(5, 2) without any assignment to an integer outputs 25.

    This is because 52 (i.e. 25) might be stored as 24.9999999 or 25.0000000001 because the return type is double. When assigned to int, 25.0000000001 becomes 25 but 24.9999999 will give output 24.
    To overcome this and output the accurate answer in integer format, we can add 0.5 to the result and typecast it to int e.g (int)(pow(5, 2)+0.5) will give the correct answer(25, in above example), irrespective of the compiler.
)`

I'd therefore ask if we could simply do 1E7 there?

Reply:
good point - done
https://github.com/quasar-team/open62541-compat/commit/4bebd8e13749b51dde90d8f1c0cc20c9e79bcae9